src/components/TodoList/TodoList.tsx

```diff
@@ -1,100 +1,128 @@
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">quis ut nam facilis et officia qui</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye-slash" />
-            </span>
-          </button>
-        </td>
-      </tr>
+export const TodoList: React.FC<Props> = ({
+  isLoadTodos,
+  whichModal,
+  query,
+  selected,
+  todoId,
+}) => {
+  const [todos, setTodos] = useState<Todo[] | null>(null);
+  const [, setFilteredTodos] = useState<Todo[] | null>(null);
+  const [selectedTodos, setSelectedTodos] = useState<Todo[] | null>(null);
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">1</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">delectus aut autem</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
+  useEffect(() => {
+    const fetchData = async () => {
+      isLoadTodos(true);
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">6</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">
-            qui ullam ratione quibusdam voluptatem quia omnis
-          </p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
+      try {
+        const todoData = await getTodos();
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">8</td>
-        <td className="is-vcentered">
-          <span className="icon" data-cy="iconCompleted">
-            <i className="fas fa-check" />
-          </span>
-        </td>
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-success">quo adipisci enim quam ut ab</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
-    </tbody>
-  </table>
-);
+        setTodos(todoData);
+        setFilteredTodos(todoData);
+        setSelectedTodos(todoData);
+      } finally {
+        isLoadTodos(false);
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  useEffect(() => {
```

Comment:
redundant useEffect, also setSelectedTodos change to useMemo

const selectedTodos = useMemo(() => {...},[todos, filtered])

Reply:
selectedTodos is already used here: const [selectedTodos, setSelectedTodos] = useState<Todo[] | null>(null); and i Cen not reuse this that way.